aries_cloudagent/commands/upgrade.py

```diff
@@ -138,65 +146,72 @@ async def add_version_record(profile: Profile, version: str):
     LOGGER.info(f"{RECORD_TYPE_ACAPY_VERSION} storage record set to {version}")
 
 
-async def upgrade(settings: dict):
+async def upgrade(
+    settings: Optional[Union[Mapping[str, Any], BaseSettings]] = None,
+    profile: Optional[Profile] = None,
+):
     """Perform upgradation steps."""
-    context_builder = DefaultContextBuilder(settings)
-    context = await context_builder.build_context()
     try:
+        if profile:
+            root_profile = profile
+            settings = profile.settings
```

Comment:
If the `settings` parameter is ignored when `profile` is provided, then maybe this function should throw an exception when both are given.